src/platforms/web/runtime/directives/show.js

```diff
@@ -53,7 +53,7 @@ export default {
     oldVnode: VNodeWithData,
     isDestroy: boolean
   ) {
-    if (!isDestroy) {
+    if (!isDestroy && el.style.display === 'none') {
```

Comment:
I think that if the style is set to anything manually via `style="display: none"` or with `:style`, it should probably keep it as well